src/main/java/frc/robot/commands/vision/NodeShifter.java

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands.vision;
+import frc.robot.Constants;
+import frc.robot.subsystems.Vision;
+import frc.robot.subsystems.DrivetrainSubsystem;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class NodeShifter extends CommandBase {
+  /** Creates a new NodeShifter. */
+  private final DrivetrainSubsystem drivetrainSubsystem;
+  private final Vision vision;
+  boolean shiftRight;
+  public NodeShifter(Vision vision, DrivetrainSubsystem drivetrainSubsystem, boolean shiftRight) {
+    // Use addRequirements() here to declare subsystem dependencies.
+    this.vision = vision;
+    this.drivetrainSubsystem = drivetrainSubsystem;
+    this.shiftRight = shiftRight;
+    addRequirements(this.vision, this.drivetrainSubsystem);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    if(vision.hasTarget() && shiftRight == true){
```

Comment:
It might be nice to create a method that checks for one of our valid tags in the scoring area instead of any valid target

Reply:
Also you might want to adopt a similar concept as your other command which is when you see a target you reset the drive pose and then you base the translation off of the drive getpose instead of the vision returned pose